docs/adr/367-branch-protection-md

```diff
@@ -0,0 +1,109 @@
+# ADR 367: Enforce Branch Protection by Default Across All Repositories
+
+## Status
+
+Accepted
+
+## Submitters
+
+- @UlisesGascon
+
+## Decision Owners
+
+- @expressjs/express-tc
+
+## Context
+
+Branch protection rules in GitHub enforce key policies—such as required reviews, status checks, and linear history—that help safeguard critical branches like `main`, `master` or version branches/tags. Despite the clear benefits, adoption across repositories has been inconsistent, leading to:
+
+- Risk of direct or accidental pushes to protected branches
+- Increased likelihood of unreviewed or unsafe code merges
+- Reduced ability to enforce standardized development workflows
+
+GitHub allows fine-grained control over branch protections, including the ability for administrators (tc members, repo captains...) to bypass rules when needed (e.g., urgent hotfixes or security patches).
+
+This decision aims to normalize a secure-by-default stance while maintaining flexibility for urgent, responsible exceptions.
+
+## Decision
+
+**Branch protection rules will be enabled by default on all repositories.**
+
+Key rules to be applied:
+
+- Protect release branches, this includes: `main`, `master`, and `v*` branches/tags
+- Require pull request reviews before merging (at least 1 approval)
+- Require all status checks to pass before merging
+
+Optional rules:
+- Enforce linear history (disable merge commits, allow squash/rebase)
+- Restrict who can push directly to the protected branches to people who can release (captains, TC, etc)
```

Comment:
Can you help me understand this optional rule and how it interacts with the exception below?

From my understanding enabling this would allow the people who can release to permanently bypass the branch protection rules, rather than temporarily. It'd be simpler to have everything follow the same process to avoid any confusion when it comes down to this.